sklearn/linear_model/coordinate_descent.py

```diff
@@ -472,6 +475,10 @@ def enet_path(X, y, l1_ratio=0.5, eps=1e-3, n_alphas=100, alphas=None,
                           ' You might want' +
                           ' to increase the number of iterations',
                           ConvergenceWarning)
+            if X.dtype == np.float32:
```

Comment:
Is this statement still valid? In any case, you should also check if `alpha` is lesser than a certain `tol`, so that your statement is valid.